lib/block_view_stake.go

```diff
@@ -0,0 +1,2600 @@
+package lib
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/golang/glog"
+	"github.com/holiman/uint256"
+	"github.com/pkg/errors"
+	"sort"
+)
+
+//
+// TYPES: StakeEntry
+//
+
+type StakeEntry struct {
+	StakeID          *BlockHash
```

Comment:
I don't think we need StakeID for anything. A StakeEntry is uniquely identified by a (ValidatorPKID, StakerPKID) pair. Feels like including this StakeID just increases risk for bugs. Going to try and delete it and see what happens.

Reply:
I deleted it, and also deleted LockedStakeID, and it seems like a very positive change. In particular, in the case of LockedStakeID, there was a place where we were confusingly setting LockedStakeID to prevStakeEntry.StakeID, which was a bit weird. Overall, the fact that we can delete this without any negative repercussions, and a slight benefit in that one place, seems to indicate that the information is unnecessarily redundant, since a (ValidatorPKID, StakerPKID) pair uniquely identifies a StakeEntry, and since a (ValidatorPKID, StakerPKID, EpochNumber) uniquely identifies a LockedStakeEntry.